src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml

```diff
@@ -16,55 +16,67 @@ limitations under the License.
         <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
-
-        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="false" isNullable="false">
-            <description>ExpectedClients</description>
+        <attribute side="server" code="0x03" define="REGISTERED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="true" isNullable="false">
+            <description>RegisteredClients</description>
             <access op="read" privilege="administer"/>
         </attribute>
-
-        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="false" isNullable="false">
+        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="true" isNullable="false">
             <description>ICDCounter</description>
             <access op="read" privilege="administer"/>
         </attribute>
+        <attribute side="server" code="0x05" define="CLIENTS_SUPPORTED_PER_FABRIC" type="INT16U" min="1" default="1" writable="false" optional="true" isNullable="false">ClientsSupportedPerFabric</attribute>
 
-
-        <command source="client" code="0x00" name="RegisterClientMonitoring" response="RegisterClientMonitoringResponse" isFabricScoped="true" optional="false">
+        <command source="client" code="0x00" name="RegisterClient" response="RegisterClientResponse" isFabricScoped="true" optional="true">
             <description> Register a client to the end device </description>
-            <arg name="ClientNodeID" type="node_id"/>
-            <arg name="Key" type="OCTET_STRING" length="16"/>
+            <arg name="CheckInNodeID" type="node_id" optional="false"/>
+            <arg name="MonitoredSubject" type="INT64U" optional="false"/>
+            <arg name="Key" type="OCTET_STRING" length="16" optional="false"/>
+            <arg name="VerificationKey" type="OCTET_STRING" length="16" optional="true"/>
             <access op="invoke" privilege="manage"/>
         </command>
 
-        <command source="server" code="0x01" name="RegisterClientMonitoringResponse" optional="false" disableDefaultResponse="true">
-            <description> RegisterClientMonitoring response command </description>
-            <arg name="Status" type="ENUM8"/>
-            <arg name="ICDCounter" type="INT32U" isNullable="true"/>
+        <command source="server" code="0x01" name="RegisterClientResponse" optional="false" disableDefaultResponse="false">
+            <description> RegisterClient response command </description>
+            <arg name="ICDCounter" type="INT32U" isNullable="false"/>
```

Comment:
```suggestion
            <arg name="ICDCounter" type="INT32U" isNullable="true"/>
```

That said, I don't understand why this is nullable in the spec, and the spec never says to set it to null...  So should it in fact not be nullable in the spec?

Reply:
Maybe i'm misunderstood something from the qualities but the ICDCounter has the Non-Volatile and Changes Omitted qualities?

Reply:
That's the _attribute_.  Here we're talking about the command field of the RegisterClientResponse command.

Reply:
Yep that's a mistake in the spec.
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7014